Bungee/src/main/java/net/minecrell/serverlistplus/bungee/BungeePlugin.java

```diff
@@ -263,18 +255,15 @@ public int getProtocolVersion() {
             } else if (favicon != null) {
                 Optional<Favicon> icon;
                 // Check if instanceof AsyncEvent for compatibility with 1.7.10
-                if (event instanceof AsyncEvent) {
-                    icon = faviconCache.getIfPresent(favicon);
-                } else {
-                    icon = faviconCache.getUnchecked(favicon);
-                }
-
-                if (icon == null) {
+                if (event instanceof AsyncEvent && !faviconCache.contains(favicon)) {
```

Comment:
`contains()` is not equivalent with a `getIfPresent()` since that will require two lookups instead of one. I think you need to add a `FaviconCache.getIfPresent()` for this to work properly.

Oh, contains() is based on getIfPresent(), well then use the result, don't discard it! :D

Also, feel free to simplify this code by dropping support for the ancient 1.7.10 case where ProxyPingEvent is not instanceof `AsyncEvent`. SLP does not support 1.7.10 anymore.

i.e. this should be something like

```java
Optional<Favicon> icon = faviconCache.getIfPresent(favicon);
if (icon == null) {
    // ...
else if (icon.isPresent()) {
    // ...
}
```

Hmm, I think this is the previous code :thinking:  Actually not sure why you changed this to a contains().

Reply:
Seeing an `Optional` being set to null was confusing to me (since the purpose behind `Optional` is to eliminate the use of null :)), so I attempted to refactor the code into something clearer. I should have explained my reasoning or put it in a separate commit, sorry about that. You're right about the the double lookup, that's clearly a performance loss.

Reply:
Ah, I understand now. Thanks for the explanation!